src/UsgsAstroLsSensorModel.cpp

```diff
@@ -443,6 +447,28 @@ std::string UsgsAstroLsSensorModel::getModelState() const {
                              m_referencePointXyz.x, m_referencePointXyz.y,
                              m_referencePointXyz.z)
 
+      state["m_sunPosition"] = json();
+      state["m_sunPosition"][0] = m_sunPosition[0];
+      state["m_sunPosition"][1] = m_sunPosition[1];
+      state["m_sunPosition"][2] = m_sunPosition[2];
+      MESSAGE_LOG(m_logger, "m_sunPosition: {} "
+                            "m_sunPosition: {} "
+                            "m_sunPosition: {} ",
+                             m_sunPosition[0],
+                             m_sunPosition[1],
+                             m_sunPosition[2])
+
+      state["m_sunVelocity"] = json();
+      state["m_sunVelocity"][0] = m_sunVelocity[0];
+      state["m_sunVelocity"][1] = m_sunVelocity[1];
+      state["m_sunVelocity"][2] = m_sunVelocity[2];
+      MESSAGE_LOG(m_logger, "m_sunVelocity: {} "
+                            "m_sunVelocity: {} "
+                            "m_sunVelocity: {} ",
+                             m_sunVelocity[0],
+                             m_sunVelocity[1],
+                             m_sunVelocity[2])

```

Comment:
These should append based on the size of the vectors. Just directly adding the first position and first velocity could cause part of the ISD to be ignored.